### PR TITLE
add power-r generator

### PR DESCRIPTION
**Pull request by pau557**
Weights follow a power-law distribution with respect to their magnitude. Good for high precision instances.

### CODE REVIEW

File: dimod/generators/random.py [hunk 1]
```diff
@@ -500,3 +500,61 @@ def doped(p: float, graph: GraphLike,
...
+            NetworkX graph.
+
+        seed: Random seed.
+
+    Returns:
+        A binary quadratic model.
+
+    """
+
+    if not isinstance(r, int):
```
Comment (arcondello) [inline dimod/generators/random.py:533]:
```
    if not isinstance(r, numbers.Integral):
```
to support integer-like types (like `np.int8`).

File: dimod/generators/random.py [hunk 2]
```diff
@@ -500,3 +500,61 @@ def doped(p: float, graph: GraphLike,
...
     return bqm
+
+
+
+@graph_argument('graph')
+def power_r(r: int, graph: GraphLike,
+          seed: Optional[int] = None) -> BinaryQuadraticModel:
+    """Generate an Ising model for a POWERr problem.
+
+    In POWERr problems all linear biases are zero and quadratic values are integers
```
Comment (arcondello) [inline dimod/generators/random.py:511]: Is there a citation for this problem class?

**Review by arcondello — COMMENTED** [review]
A few minor nits but otherwise looking good!

**Comment by arcondello** [comment]
Oh, also needs a [release note](https://github.com/dwavesystems/dimod?tab=readme-ov-file#release-notes). Probably something like `Add ``power_r()`` generator function to ``dimod.generators``. The weights of generated Ising problems follow a power-law distribution with respect to their magnitude.`

File: dimod/generators/random.py (hunk 2 above)
Comment (pau557) [reply dimod/generators/random.py:513]: not yet!

**Comment by pau557** [comment]
> Oh, also needs a [release note](https://github.com/dwavesystems/dimod?tab=readme-ov-file#release-notes). Probably something like `` Add ``power_r()`` generator function to ``dimod.generators``. The weights of the generated Ising problems follow a power-law distribution with respect to their magnitude. ``

**Comment by pau557** [comment]
Getting an error with `reno`. Will try again tomorrow

File: dimod/generators/random.py [hunk 3]
```diff
@@ -500,3 +502,63 @@ def doped(p: float, graph: GraphLike,
         bqm.add_interaction(u, v, J)
 
     return bqm
+
+
+
+@graph_argument('graph')
+def power_r(r: int, graph: GraphLike,
+          seed: Union[None, int, np.random.Generator] = None,) -> BinaryQuadraticModel:
```
Comment (arcondello) [inline dimod/generators/random.py:510]: Careful with the indentation. We should also make `seed` a keyword only argument.

```suggestion
def power_r(r: int, 
            graph: GraphLike,
            *,
            seed: Union[None, int, np.random.Generator] = None,
            ) -> BinaryQuadraticModel:
```

Comment (pau557) [reply dimod/generators/random.py:510]: is it ok to use the formatting that Black suggests instead?

```python
@graph_argument("graph")
def power_r(
    r: int,
    graph: GraphLike,
    *,
    seed: Union[None, int, np.random.Generator] = None,
) -> BinaryQuadraticModel:


```

Comment (arcondello) [reply dimod/generators/random.py:510]: Hah, technically a pep8 violation (https://github.com/psf/black/issues/1178 for the rabbit hole), but sure, that's OK.

**Review by arcondello — APPROVED** [review]

**Pull request merged**